src/main/java/mekanism/common/item/gear/ItemMekaTool.java

```diff
@@ -342,18 +360,50 @@ public boolean canApplyAtEnchantingTable(ItemStack stack, Enchantment enchantmen
         return false;
     }
 
-    @Override
-    public boolean supportsSlotType(ItemStack stack, @Nonnull EquipmentSlot slotType) {
-        return IModeItem.super.supportsSlotType(stack, slotType) && getModules(stack).stream().anyMatch(Module::handlesModeChange);
-    }
+    public enum MekaToolMode implements IRadialSelectorEnum<MekaToolMode>, IHasTextComponent {
+        ONE("mode_1"),
+        TWO("mode_2"),
+        THREE("mode_3"),
+        FOUR("mode_4"),
+        FIVE("mode_5"),
+        SIX("mode_6");
 
-    @Override
-    public void changeMode(@Nonnull Player player, @Nonnull ItemStack stack, int shift, boolean displayChangeMessage) {
-        for (Module<?> module : getModules(stack)) {
-            if (module.handlesModeChange()) {
-                module.changeMode(player, stack, shift, displayChangeMessage);
-                return;
-            }
+        private final String key;
+
+        MekaToolMode(String key) {
+            this.key = key;
+        }
+
+        public String getKey() {
+            return key;
+        }
+
+        public boolean isEnabled() {
+            return true;
+        }
+
+        public static MekaToolMode byIndexStatic(int index) {
```

Comment:
What is the point of having a static and non-static method for doing the same thing?

Reply:
So I'm not sure since I'm not at my pc right now and I copied some of this from the atomic disassembler implementation. But I think that the instance method is an override for the IRadialModeEnum, but the static one is the only one used I think.

Reply:
So I'm not sure since I'm not at my pc right now and I copied some of this from the atomic disassembler implementation. But I think that the instance method is an override for the IRadialModeEnum, but the static one is the only one used I think.

Reply:
Double checked and pretty much every mode object has this exact implementation as well, including the valve blocks. I can't figure out how the instance one could be used where the static wouldn't. Maybe if you wanted the next mode but didn't know which enum you had?